### ISSUE TITLE
album library not properly updating

### ISSUE THREAD

**Issue by alan12345678** [quodlibet/quodlibet#1917]
I'm presently (short term only) accessing a music library via a Windows share with its attached external HDD, so it is possible that some of my problem is due to mount/share issues.

The music library - as presently viewed via a second "album list" browser in QL - doesn't appear to be properly reflecting changes to albums/tracks when they are edited on the Windows host machine.

Due to the way my library was organized under Windows, I have numbers of duplicate songs, sometimes in other formats, and which individual songs live in a "general" directory outside of complete albums (which have folder-per-album). I therefore have a direct comparison in the album list browser where certain instances will show one version of a song correctly labeled with both album and artist displayed, and another version where only album name is displayed.

I open a tag editor on the Windows host, and by loading all versions of the song together, can tag everything identically. Once this is done, even a reboot and then "rebuild" library in QL fails to display the changes. Looking at the information panel in the album view, it displays the pre-edit condition of the  songs, and not their newly edited status with complete tag information.

I don't know where QL stores its music database. Can I delete it and start again? Can I manually edit it? Can I somehow "force" a complete rebuild? (would prefer the first option).

**Comment by lazka** [comment]
Which other tag editor are you using?

**Comment by lazka** [comment]
btw. rebuilding the library will recreate the "Music database", so all tags will be thrown away, except ratings and playback statistics.

**Comment by alan12345678** [comment]
mp3tag

Please note that I have mp3tag set up to use ID3v2.3 UTF16, because using v2.4 UTF 8, attached cover art doesn't show up in Windows.

I have tried to do such things as add specific "artistsort" tags as a way to help QL derive ~performer information, but no joy as yet...

**Comment by alan12345678** [comment]
should also add that I would happily do the tagging within QL using the musicbrainz plugin, but as above, my share permissions won't allow the file overwrite (despite being logged in with the correct credentials)

**Comment by lazka** [comment]
Can you send me one (maybe small) affected file per mail? reiter.christoph@gmail.com

> I'm presently (short term only) accessing a music library via a Windows share with its attached external HDD, so it is possible that some of my problem is due to mount/share issues.

I do most of my testing with a share as well, so that shouldn't be an issue, at least in general.

> Please note that I have mp3tag set up to use ID3v2.3 UTF16, because using v2.4 UTF 8, attached cover art doesn't show up in Windows.

QL will always write v2.4 btw

> but as above, my share permissions won't allow the file overwrite

Do you mean QL fails to write to the files but should succeed?

**Comment by alan12345678** [comment]
will do. I do very much appreciate the speedy and effective response, btw ;0)

I might point out now that I have a new internal HDD on order, and when it arrives I'll be transferring my library locally, so don't want to think this is a critical issue. That said, I do learn a lot about linux and troubleshooting :0)

As for the permissions: tbh I'm still a bit new to all this. I've tried many different permutations of the basic cifs mount in fstab, but have devolved back to the most basic and simple. As such, the share mounts using credentials which are the exact ones I set the share up on the host i.e. same username, pwd and domain. The library is shared with "full" permissions. Whether this utlimately means that QL *should* have full permissions I can't really tell you.

**Comment by alan12345678** [comment]
can I attach files and send via this email here??
 
*alan ansell figurative painting Itzac France*
He would be a poet who could impress the winds and streams into his
service, to speak for him.
- Henry David Thoreau
 
 
----- Original message -----

**Comment by lazka** [comment]
> can I attach files and send via this email here??

Not sure. You can DnD files in the web interface at least. If that was an attempt, apparently it doesn't work :)

----

So you are using QL only on Fedora only? I somehow assumed you were using Windows..

**Comment by alan12345678** [comment]
no it wasn't. No probs, I will attempt to d&d on the web page. I
asked b/c I was using a different player (also on github) and
couldn't - possibly due to his account settings. This is email, I'll
try via the webpage
 
*alan ansell figurative painting Itzac France*
He would be a poet who could impress the winds and streams into his
service, to speak for him.
- Henry David Thoreau
 
 
----- Original message -----

**Comment by alan12345678** [comment]
can't attach music files here (makes sense I suppose). Can I send them to you using Box?

**Comment by lazka** [comment]
OK, whatever you prefer. Normal email attachment works too.

**Comment by alan12345678** [comment]
so:

Just installed a new internal HDD, mounted it and copied across my library. I'm still seeing this curious behaviour where it treats different file types differently and does not display "artist" name in the album browser.

I note that if I open an information pane for two different file types - one of which displays correctly - the file type(s) missing certain information can be shown to have no "performer" value extracted, despite these same files being tagged with at least the same information as, for example, an mp3 - which does display okay.

Now that the music files are local, I've used plugins to lookup musicbrainz, plus run various tools to check tagging, file lengths etc. Cannot find problems with the files themselves, but I do have this discrepancy in the way m4a, wma and flac are treated differently from mp3

**Comment by alan12345678** [comment]
All completely sorted. Thanks! Reinstalled Fedora completely. Learned important lesson: DO NOT mix use of dnf (terminal) and gnome packaging, because they don't use the same installation db!
Player is working brilliantly, have spent a lot of time understanding tags and library management. Superb!!

(now trying to discover how to edit the album download plugin to add other sources)

**Comment by GaryReg** [comment]
I have (what I think) is a similar problem. I first installed QuodLibet (QL) on my iMac (Sierrra 10.12.6) and it found all my music on my Boot drive under iTunes/Music, when it started up. I moved my library to another drive and now cannot get QL to find the songs again. I tried adding folders, files, etc. Restarted iMac and no go. I deleted the App, downloaded the newest Version (4.02). Restarted iMac and still no go.  I'm not a programmer so don't understand scrips and tagging. Shouldn't QL be able to recognize Music files when it Scans?
